src/devices/src/virtio/rng/device.rs

```diff
@@ -404,9 +483,135 @@ mod tests {
             ],
         );
 
+        let entropy_event_fails = METRICS.entropy.entropy_event_fails.count();
+        let entropy_event_count = METRICS.entropy.entropy_event_count.count();
+        let entropy_bytes = METRICS.entropy.entropy_bytes.count();
+        let host_rng_fails = METRICS.entropy.host_rng_fails.count();
         assert_eq!(th.emulate_for_msec(100).unwrap(), 1);
-        assert_eq!(METRICS.entropy.entropy_event_fails.count(), 1);
-        assert_eq!(METRICS.entropy.entropy_event_count.count(), 4);
-        assert_eq!(METRICS.entropy.entropy_bytes.count(), 542);
+        assert_eq!(
+            METRICS.entropy.entropy_event_fails.count(),
+            entropy_event_fails
+        );
+        assert_eq!(
+            METRICS.entropy.entropy_event_count.count(),
+            entropy_event_count + 1
+        );
+        assert_eq!(METRICS.entropy.entropy_bytes.count(), entropy_bytes + 512);
+        assert_eq!(METRICS.entropy.host_rng_fails.count(), host_rng_fails);
+    }
+
+    #[test]
+    fn test_bad_rate_limiter_event() {
+        let mem = create_virtio_mem();
```

Comment:
Nit: Could we use `single_region_mem` from the same `test_utils` module instead here?

Reply:
Fair, I guess the prefered solution would then be to update all other tests to use `create_virtio_mem` and that's not in-scope here 👍